File: eel-core/src/main/kotlin/io/eels/component/jdbc/JdbcPrimitives.kt
package io.eels.component.jdbc

import io.eels.schema.Schema
import io.eels.util.Logging
import java.sql.Connection
import java.sql.DriverManager
import java.sql.ResultSet

interface JdbcPrimitives : Logging {

  fun connect(url: String): Connection {
    logger.info("Connecting to jdbc source $url...")
    val conn = DriverManager.getConnection(url)
    logger.debug("Connected to $url")
    return conn
  }

  fun schemaFor(dialect: JdbcDialect, rs: ResultSet): Schema {
    val schema = JdbcSchemaFn(rs, dialect)
    logger.debug("Fetched schema:\n" + schema.show())
    return schema
  }
}

File: eel-core/src/main/scala/io/eels/component/jdbc/JdbcPrimitives.scala
package io.eels.component.jdbc

import java.sql.{Connection, DriverManager, ResultSet}

import com.sksamuel.exts.Logging
import io.eels.component.jdbc.dialect.JdbcDialect
import io.eels.schema.StructType

trait JdbcPrimitives extends Logging {

  def connect(url: String): Connection = {
    logger.info("Connecting to jdbc source $url...")
    val conn = DriverManager.getConnection(url)
    logger.debug("Connected to $url")
    conn
  }

  def schemaFor(dialect: JdbcDialect, rs: ResultSet): StructType = {
    val schema = JdbcSchemaFns.fromJdbcResultset(rs, dialect)
    logger.debug("Fetched schema:\n" + schema.show())
    schema
  }
}
